### PR TITLE
Add editable description to the entity type editor page

### PR DESCRIPTION
**Pull request by luisbettencourt**
## 🌟 What is the purpose of this PR?

This PR adds a description field to the entity type editor page, similarly to how it was implemented in blockprotocol.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204274556967336/1204485873929278/f) _(internal)

## 🔍 What does this change?

- Used `EditableField` from `@hashintel/block-design-system` for the description field.

### 🚢 Has this modified a publishable library?

This PR:

 - [x] does not modify any publishable blocks or libraries, or modifications do not need publishing 

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] is internal and does not require a docs change 

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] does not affect the execution graph

## ❓ How to test this?

1.  Create an entity type;
2.  Edit and publish description;
3.  Make sure it persists through reload;

## 📹 Demo

demo

### CODE REVIEW

**Comment by luisbettencourt** [comment]
@nonparibus @CiaranMn It might make more sense to have the description field under the definition tab, instead of in the header. Please LMK if you want me to move it.

**Comment by nonparibus** [comment]
@luisbettencourt I'd keep it where it is, but simply remove the "Description" label.

File: apps/hash-frontend/src/pages/[shortname]/types/entity-type/[...slug-maybe-version].page/entity-type-description.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    control,
+    name: "description",
+  });
+
+  return (
+    <>
+      <Typography variant="h5" sx={{ mb: 1, fontWeight: 500 }}>
+        Description
+      </Typography>
+
```
Comment (yusufkinatas) [inline apps/hash-frontend/src/pages/[shortname]/types/entity-type/[...slug-maybe-version].page/entity-type-description.tsx:25]: Let's remove this as David suggests

**Review by yusufkinatas — CHANGES_REQUESTED** [review]

File: apps/hash-frontend/src/pages/[shortname]/types/entity-type/[...slug-maybe-version].page/entity-type-description.tsx (hunk 1 above)
Comment (luisbettencourt) [reply apps/hash-frontend/src/pages/[shortname]/types/entity-type/[...slug-maybe-version].page/entity-type-description.tsx]: Done in b6558f827674f7447c5d807aed6fa04d7afce5f2

**Review by yusufkinatas — APPROVED** [review]

**Pull request merged**